flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/proxy/KinesisProxy.java

```diff
@@ -186,7 +186,10 @@ protected KinesisProxy(Properties configProps) {
 	 * @return
 	 */
 	protected AmazonKinesis createKinesisClient(Properties configProps) {
-		return AWSUtil.createKinesisClient(configProps, new ClientConfigurationFactory().getConfig());
+
+		ClientConfiguration awsClientConfig = new ClientConfigurationFactory().getConfig();
+		AWSUtil.setAwsClientConfigProperties(awsClientConfig, configProps);
+		return AWSUtil.createKinesisClient(configProps, awsClientConfig);
```

Comment:
Currently, the explicit Flink-defined keys in `AWSConfigConstants` will override whatever is set via the generic mechanism, correct?

Reply:
That's correct, the generic property setting takes place before those keys are processed.

Reply:
Actually there does not seem to be any overlap. `AWSConfigConstants.AWS_REGION` and `AWSConfigConstants.AWS_ENDPOINT` are used to construct the client, not the client config.